client/components/deposits-overview/utils.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { sprintf } from '@wordpress/i18n';
+import moment from 'moment';
+import interpolateComponents from '@automattic/interpolate-components';
+
+/**
+ * Internal dependencies
+ */
+import { formatCurrency } from 'utils/currency';
+import strings from './strings';
+import { getDepositMonthlyAnchorLabel } from 'wcpay/deposits/utils';
+
+type NextDepositTableData = {
+	date: number;
+	status: string;
+	amount: string;
+};
+
+/**
+ * Formats the next deposit data from the overview object into an object that can be used in the Next Deposits table.
+ *
+ * @param {AccountOverview.Overview} overview - The overview object containing information about the next scheduled deposit.
+ * @return {NextDepositTableData} An object containing the formatted next deposit data, with the following properties:
+ * - date: A Unix timestamp representing the date of the next scheduled deposit.
+ * - status: A string representing the status of the next scheduled deposit. If no status is provided, defaults to 'estimated.
+ * - amount: A formatted string representing the amount of the next scheduled deposit in the currency specified in the overview object.
+ */
+export const getNextDeposit = (
+	overview: AccountOverview.Overview
+): NextDepositTableData => {
+	if ( ! overview ) {
+		return {
+			date: 0,
+			status: 'estimated',
+			amount: formatCurrency( 0, 'USD' ),
+		};
+	}
+
+	const { currency, nextScheduled } = overview;
+
+	return {
+		date: nextScheduled.date ?? 0,
+		status: nextScheduled.status ?? 'estimated',
+		amount: formatCurrency( nextScheduled.amount ?? 0, currency ),
+	};
+};
+
+/**
+ * Fetches the description for the deposit schedule from the account object.
+ *
+ * @param {AccountOverview.Account} account The account object containing information about the deposit schedule.
+ * @return {JSX.Element} The description for the deposit schedule.
+ */
+export const getDepositScheduleDescription = (
+	account: AccountOverview.Account
+): JSX.Element => {
+	const schedule = account.deposits_schedule;
```

Comment:
Regarding naming (continuing discussion from slack): this could be a regular function component. Also, it looks like it only needs schedule, not the whole account. You might be able to simplify props more by passing only what's needed from `schedule`. Maybe it's more typescript style to pass the smallest "struct" needed, e.g. `schedule`?

Something like: 

```jsx
// Renders a deposit schedule as a descriptive rich-text sentence for display in UI. 
function DepositScheduleDescription( { schedule } ) {
  return (…); 
}

<DepositScheduleDescription schedule={schedule} />
```